src/main/java/eu/europeana/api/recommend/service/RecommendService.java

```diff
@@ -75,11 +84,151 @@ public Mono getRecommendationsForRecord(String recordId, int pageSize, String to
+        StringBuilder setApiUrl = new StringBuilder(config.getSetApiEndpoint()).
+                append(getRecommendServiceUtils().setApiSearchQuery(entityId, config.getSetApiPageSize(), apikey));
+        getItems(request, setApiUrl.toString());
+
+        // serialise the request
+        return serialiseEntityRequest(request);
+    }
+
+    /**
+     * Sets the labels in the EntityRecommendRequest
+     * extracts the prefLabel and altLabel from entity api json response
+     *
+     * @param request
+     * @param entityUrl
+     */
+    private void getLabels(EntityRecommendRequest request, String entityUrl) {
+        try {
+            JSONObject jsonObject = new JSONObject(getEntityApiSearchResponse(entityUrl));
+            List<String> extractedLabels = getRecommendServiceUtils().extractLabels(jsonObject);
+            List<Labels> labels = new ArrayList<>();
+            if (!extractedLabels.isEmpty()) {
+                for (String label : extractedLabels) {
+                    labels.add(new Labels(label));
+                }
+                request.setLabels(labels);
+            } else {
+                labels.add(new Labels(""));
+                request.setLabels(labels);
+            }
+        } catch (JSONException e) {
+            LOG.error("Error parsing entity api response for url {}", entityUrl);
+        }
+    }
+
+    /**
+     * Sets the items in the EntityRecommendRequest.
+     * Extracts the items from the set api json response
+     *
+     * @param request
+     * @param setApiUrl
+     */
+    private void getItems(EntityRecommendRequest request, String setApiUrl) {
+        try {
+            JSONObject jsonObject = new JSONObject(getSetApiSearchResponse(setApiUrl));
+            List<String> items=getRecommendServiceUtils().extractItems(jsonObject);
+            if (!items.isEmpty()) {
+                request.setItems(items.toArray(new String[0]));
+            } else {
+                request.setItems(new String[]{""});
+            }
+        } catch (JSONException e) {
+            LOG.error("Error parsing set api response for url {}", setApiUrl);
+        }
+    }
+
+    /**
+     * Build the entity id/url
+     *
+     * @param type
+     * @param id
+     * @return
+     */
+    public String buildEntityId(String type, String id) {
+        StringBuilder entityId = new StringBuilder("http://data.europeana.eu");
+        entityId.append("/").append(type);
+        entityId.append("/base/").append(id);
+        return entityId.toString();
+    }
+
+    /**
+     * Serialises the EntityRecommendRequest
+     * @param request
+     * @return
+     */
+    private String serialiseEntityRequest(EntityRecommendRequest request) {
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            return mapper.writeValueAsString(request);
+        } catch (JsonProcessingException e) {
+            LOG.error("Error serialising the Entity recommendation request");
```

Comment:
Can we also log the JSONException as well (and fix the SQ warning that we're ignoring it)?

Reply:
done